examples/shader/post_processing.rs

```diff
@@ -0,0 +1,340 @@
+            dimension: TextureDimension::D2,
+            format: TextureFormat::Bgra8UnormSrgb,
+            mip_level_count: 1,
+            sample_count: 1,
+            usage: TextureUsages::TEXTURE_BINDING
+                | TextureUsages::COPY_DST
+                | TextureUsages::RENDER_ATTACHMENT,
+        },
+        ..default()
+    };
+
+    // fill image.data with zeroes
+    image.resize(size);
+
+    let image_handle = images.add(image);
+
+    let cube_handle = meshes.add(Mesh::from(shape::Cube { size: 4.0 }));
+    let cube_material_handle = materials.add(StandardMaterial {
+        base_color: Color::rgb(0.8, 0.7, 0.6),
+        reflectance: 0.02,
+        unlit: false,
+        ..default()
+    });
+
+    // This specifies the layer used for the first pass, which will be attached to the first pass camera and cube.
+    let first_pass_layer = RenderLayers::layer(1);
+
+    // The cube that will be rendered to the texture.
+    commands
+        .spawn_bundle(PbrBundle {
+            mesh: cube_handle,
+            material: cube_material_handle,
+            transform: Transform::from_translation(Vec3::new(0.0, 0.0, 1.0)),
+            ..default()
+        })
+        .insert(FirstPassCube)
+        .insert(first_pass_layer);
+
+    // Light
+    // NOTE: Currently lights are shared between passes - see https://github.com/bevyengine/bevy/issues/3462
+    commands.spawn_bundle(PointLightBundle {
+        transform: Transform::from_translation(Vec3::new(0.0, 0.0, 10.0)),
+        ..default()
+    });
+
+    // First pass camera
+    let render_target = RenderTarget::Image(image_handle.clone());
+    clear_colors.insert(render_target.clone(), Color::WHITE);
+    commands
+        .spawn_bundle(PerspectiveCameraBundle::<FirstPassCamera> {
+            camera: Camera {
+                target: render_target,
+                ..default()
+            },
+            transform: Transform::from_translation(Vec3::new(0.0, 0.0, 15.0))
+                .looking_at(Vec3::default(), Vec3::Y),
+            ..PerspectiveCameraBundle::new()
+        })
+        .insert(first_pass_layer);
+    // NOTE: omitting the RenderLayers component for this camera may cause a validation error:
+    //
+    // thread 'main' panicked at 'wgpu error: Validation Error
+    //
+    //    Caused by:
+    //        In a RenderPass
+    //          note: encoder = `<CommandBuffer-(0, 1, Metal)>`
+    //        In a pass parameter
+    //          note: command buffer = `<CommandBuffer-(0, 1, Metal)>`
+    //        Attempted to use texture (5, 1, Metal) mips 0..1 layers 0..1 as a combination of COLOR_TARGET within a usage scope.
+    //
+    // This happens because the texture would be written and read in the same frame, which is not allowed.
+    // So either render layers must be used to avoid this, or the texture must be double buffered.
+
+    let quad_size = 400.0;
+    let quad_handle = meshes.add(Mesh::from(shape::Quad::new(Vec2::splat(quad_size))));
```

Comment:
that's pretty ugly, but I didn't know exactly how to make it better without adding too much code

Reply:
- best would be to have a single triangle I guess ?
- second best woul be to have a quad the size of the screen ?